cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-sync.php

```diff
@@ -172,13 +172,58 @@ public function get_signature( $post_id ) {
 		return $return;
 	}
 
+	/**
+	 * Generate a new Public ID for an asset.
+	 *
+	 * @param int $attachment_id The attachment ID for the new public ID.
+	 *
+	 * @return string|null
+	 */
+	public function generate_public_id( $attachment_id, $prefix = null ) {
+		$settings   = $this->plugin->config['settings'];
+		$cld_folder = trailingslashit( $settings['sync_media']['cloudinary_folder'] );
+		$file       = get_attached_file( $attachment_id );
+		$file_info  = pathinfo( $file );
+		$public_id  = $cld_folder . $prefix . $file_info['filename'];
+		// Get the type.
+		if ( wp_attachment_is( 'image', $attachment_id ) ) {
+			$type = 'image';
+		} elseif ( wp_attachment_is( 'video', $attachment_id ) ) {
+			$type = 'video';
+		} elseif ( wp_attachment_is( 'video', $attachment_id ) ) {
```

Comment:
Don't we have here a typo? I think that the previous check was already for `video`.